### ISSUE TITLE
Modifying (`.add()`, `.subtract()`, etc.) a  dayjs instance created in UTC corrupts it

### ISSUE THREAD

**Issue by marnixhoh** [iamkun/dayjs#2037]
`dayjs.tz(0, 'UTC').format()`

`'1970-01-01T00:00:00Z'` - Result
`'1970-01-01T00:00:00Z'` - Expected
___

`dayjs.tz(0, 'UTC').add(0,'minutes').format()`

`'1969-12-31T23:00:00Z'` - Result
`'1970-01-01T00:00:00Z'` - Expected
___

`dayjs.tz(0, 'UTC').add(1,'minutes').format()`

`'1969-12-31T23:01:00Z'` - Result
`'1970-01-01T00:01:00Z'` - Expected

**Information**
 - Day.js Version 1.11.5
 - OS: Mac
 - Browser: Chrome

**Comment by BePo65** [comment]
I could reproduce the issue; working on a solution / pr.

**Comment by maulik1729** [comment]
@BePo65 found any solution for this? 
Thanks.

**Comment by BePo65** [comment]
Yes I tried every combination and found that the problem is the utc time zone (even as gmt e.a.) I am working on it for the last 2 weeks, but i could not yet solve all combinations. At the moment I just have to fix the .tz('UTC') variant.

Please be patient as it will take me for sure a few more days; but I won't surrender 😃

**Comment by maulik1729** [comment]
`dayjs(data).tz('UTC').format('YYYY-MM-DD HH:mm:ss'): incorrect output`
`dayjs(data).utc().format('YYYY-MM-DD HH:mm:ss'): correct output`
@BePo65 this gives different output shouldn't we use utc implementation when timezone has offset of zero?
`proto.tz = function (timezone = defaultTimezone, keepLocalTime) {
        const oldOffset = this.utcOffset();
        const date = this.toDate();
        const target = date.toLocaleString("en-US", { timeZone: timezone });
        const diff = Math.round((date - new Date(target)) / 1000 / 60);
        const offset = -Math.round(date.getTimezoneOffset() / 15) * 15 - diff;
        if(offset === 0){                           
            return this.utc(keepLocalTime);    //Please check this    
        }
        let ins = d(target)
            .$set(MS, this.$ms)
            .utcOffset(offset, true);
        if (keepLocalTime) {
            const newOffset = ins.utcOffset();
            ins = ins.add(oldOffset - newOffset, MIN);
        }
        ins.$x.$timezone = timezone;
        return ins;
    };`

**Comment by BePo65** [comment]
good idea; trying to use it in my pr.
